#!/usr/bin/env python
PACKAGE = "aggregator"
# The following line should not be here for Groovy and above
import roslib;roslib.load_manifest(PACKAGE)

# For Groovy and above, the following line should be
# from dynamic_reconfigure.parameter_generator_catkin import *
from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add("depth_PID", bool_t, 0, "Manual-0 or Auto-1", False)
gen.add("heading_PID", bool_t, 0, "Manual-0 or Auto-1", False)
gen.add("forward_PID", bool_t, 0, "Manual-0 or Auto-1", False)
gen.add("backward_PID", bool_t, 0, "Manual-0 or Auto-1", False)
gen.add("sidemove_PID", bool_t, 0, "Manual-0 or Auto-1", False)
gen.add("teleop", bool_t, 0, "Manual-0 or Auto-1", False)
#gen.add("reset", bool_t, 0, "Manual-0 or Auto-1", False)

gen.add("depth_kp", double_t, 0, "P const for depth controller", 6500, 0, 10000)
gen.add("depth_ki", double_t, 0, "I const for depth controller", 600, 0, 10000)
gen.add("depth_kd", double_t, 0, "D const for depth controller", 50, 0, 10000)
gen.add("depth_setpoint", double_t, 0, "D const for depth controller", 0.5, -2, 2)

gen.add("heading_kp", double_t, 0, "P const for heading controller", 20, 0, 10000)
gen.add("heading_ki", double_t, 0, "I const for heading controller", 0, 0, 10000)
gen.add("heading_kd", double_t, 0, "D const for heading controller", 0, 0, 10000)
gen.add("heading_setpoint", double_t, 0, "D const for heading controller", 0, 0, 360)

gen.add("forward_kp", double_t, 0, "P const for forward controller", 0, 0, 10000)
gen.add("forward_ki", double_t, 0, "I const for forward controller", 0, 0, 10000)
gen.add("forward_kd", double_t, 0, "D const for forward controller", 0, 0, 10000)
#gen.add("forward_setpoint", double_t, 0, "D const for forward controller", 0, 0, 1)
#gen.add("forward_input", double_t, 0, "D const for forward controller", 0, 0, 1)

gen.add("backward_kp", double_t, 0, "P const for backward controller", 0, 0, 10000)
gen.add("backward_ki", double_t, 0, "I const for backward controller", 0, 0, 10000)
gen.add("backward_kd", double_t, 0, "D const for backward controller", 0, 0, 10000)
gen.add("backward_setpoint", double_t, 0, "D const for backward controller", 0, 0, 1)

gen.add("sidemove_kp", double_t, 0, "P const for sidemove controller", 0, 0, 10000)
gen.add("sidemove_ki", double_t, 0, "I const for sidemove controller", 0, 0, 10000)
gen.add("sidemove_kd", double_t, 0, "D const for sidemove controller", 0, 0, 10000)
#gen.add("sidemove_setpoint", double_t, 0, "D const for sidemove controller", 0, 0, 1)

#gen.add("ratio_t1", double_t, 0, "thruster 1 ratio", 1, 0, 1)
#gen.add("ratio_t2", double_t, 0, "thruster 2 ratio", 1, 0, 1)
#gen.add("ratio_t3", double_t, 0, "thruster 3 ratio", 1, 0, 1)
#gen.add("ratio_t4", double_t, 0, "thruster 4 ratio", 1, 0, 1)
#gen.add("ratio_t5", double_t, 0, "thruster 5 ratio", 1, 0, 1)
#gen.add("ratio_t6", double_t, 0, "thruster 6 ratio", 1, 0, 1)

exit(gen.generate(PACKAGE, "aggregator", "controller_param"))
